### PR TITLE
Upgrade spring-cloud-dependencies version to 2021.0.0

### PR DESCRIPTION
**Pull request by moarychan**
Changelog:
- Upgrade spring-cloud-dependencies version to 2021.0.0
- Add new version tags in dependencies file

### CODE REVIEW

File: sdk/spring/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 4.0.0-beta.3 (Unreleased)
 
 ### Features Added
+- Support spring-cloud-dependencies version to 2021.0.0.
```
Comment (chenrujun) [inline sdk/spring/CHANGELOG.md:6]: How about write like this:

```
## 4.0.0-beta.3 (Unreleased)

 - Supported spring-boot version: 2.6.0 - 2.6.1
 - Supported spring-cloud version: 2021.0.0

### Features Added

### Breaking Changes

### Dependency Updates
Upgrade dependency according to spring-boot-dependencies:2.6.1 and spring-cloud-dependencies:2021.0.0

```

**Comment by moarychan** [comment]
Hi @chenrujun , please help review, thanks!

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by chenrujun** [comment]
> @moarychan
Temporarily fix compilation errors, this class will be removed in other PR.

@moarychan , could you please paste the link of that PR?

**Comment by moarychan** [comment]
https://github.com/Azure/azure-sdk-for-java/pull/26170

**Pull request merged**